STYLE_GUIDE.md

```diff
@@ -0,0 +1,185 @@
+
+## Accessibility
+
+We aim to meet [WCAG 2.1 Level AA](https://www.w3.org/WAI/standards-guidelines/wcag/) standards where possible, while acknowledging that this is a data visualization, so some criteria will be difficult to pass without extraordinary effort. However we will do our best within reason.
+
+Kedro-Viz should be navigable with different input devices (e.g. mouse, keyboard, and touchscreen), with obvious and visible focus states on all interactive elements. Text elements should have sufficient colour contrast and font-size to ensure readability to a minimum Level AA standard. Use [WAI-ARIA](https://www.w3.org/WAI/standards-guidelines/aria/) features where applicable to improve screenreader support.
+
+We recommend checking [axe](https://www.deque.com/axe/) and [Lighthouse](https://developers.google.com/web/tools/lighthouse) to review for potential accessibility errors if unsure.
+
+## Semantic HTML
+
+Ensure that page content is marked up with [semantic HTML elements](https://html.com/semantic-markup/) in order to ensure that it is as usable and accessible as possible.
+
+Avoid:
+```jsx
+<div className="form">
+  <div>Contact us</div>
+  <div>Use this form to send us a message.</div>
+  <div>Email address:</div>
+  <input type="text" />
+  <div onClick={handleSubmit}>Submit</div>
+</div>
+```
+Prefer:
+```jsx
+<form onSubmit={handleSubmit}>
+  <h1>Contact us</h1>
+  <p>Use this form to send us a message.</p>
+  <label htmlFor="email">Email address:</label>
+  <input id="email" type="email" />
+  <button>Submit</button>
+</form>
+```
+
+## CSS
+
+### Class names
+
+This project uses the [BEM naming convention](http://getbem.com/), to avoid conflicts, reduce rule specificity, and make classes easier to read and understand. [Reducing rule specificity](https://css-tricks.com/strategies-keeping-css-specificity-low/) is particularly important for improving CSS maintainability, so we try to avoid nesting CSS selectors as much as possible.
+
+We use the `pipeline-` prefix for most HTML/CSS classes, to reduce the risk of clashes in the global namespace when this project is imported into other applications. It's important never 
+
+<!-- @TODO Explain usage of the 'kedro' class, Kedro-UI, and webfonts -->
+
+### Variables
+
+We use Sass variables to declare colours and reused values wherever useful. It's not necessary to declare every CSS unit value as a variable, but if a value is used in multiple places (for related reasons), then it makes sense to make it a variable - either locally in the component, or application-wide in `src/styles/_variables.scss`. Using a named variable instead of a number can clarify your intent, and help you avoid having to update a value in multiple places later on.
```

Comment:
```suggestion
We use Sass variables to declare colours and reuse values wherever useful. It's not necessary to declare every CSS unit value as a variable, but if a value is used in multiple places (for related reasons), then it makes sense to make it a variable - either locally in the component, or application-wide in `src/styles/_variables.scss`. Use a named variable instead of a number: it can clarify your intent, and help you avoid updates to the value in multiple places later on.
```